Assets/MRTK/Providers/LeapMotion/Editor/ConfigurationChecker/LeapMotionConfigurationChecker.cs

```diff
@@ -0,0 +1,389 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.﻿
+
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.LeapMotion
+{
+    /// <summary>
+    /// Class that checks if the Leap Motion Core assets are present and configures the project if they are.
+    /// </summary>
+    [InitializeOnLoad]
+    static class LeapMotionConfigurationChecker
+    {
+        // The presence of the LeapXRServiceProvider.cs is used to determine if the Leap Motion Core Assets are in the project.
+        private const string trackedLeapFileName = "LeapXRServiceProvider.cs";
+        private static readonly string[] Definitions = { "LEAPMOTIONCORE_PRESENT" };
+
+        // True if the Leap Motion Core Assets are in the project.
+        private static bool isLeapInProject = false;
+
+        // The path difference between the root of assets and the root of the Leap Motion Core Assets.
+        private static string pathDifference = "";
+
+        // Array of paths to Leap Motion testing directories that will be removed from the project.
```

Comment:
Nit: If directories should end in /, should also add a comment saying that (i.e. otherwise a new person might just add:

LeapMotion/Core/Player/Tests

Reply:
Added comment making sure that a test directory ends with '/'

Reply:
Alternatively, you can specify that you don't need the '/' - I *think* the code supports this as well.